app/js/components/names.jsx

```diff
@@ -1,265 +1,241 @@
-            preferred: true,
-            creator: '',
-            dateCreated: '',
-            res: '',
-        };
-        this.handleEdit = this.handleEdit.bind(this);
-        this.handleSave = this.handleSave.bind(this);
-        this.handleCancel = this.handleCancel.bind(this);
-        this.handleChange = this.handleChange.bind(this);
-        this.fetch = this.fetch.bind(this);
-    }
-
-    /**
-     *
-     * set to state patient attributes
-     *
-     * @memberOf Names
-     */
-    componentWillMount() {
-        this.fetch();
-    };
-
-    /**
-     *
-     * make API call to get the values then set to state
-     *
-     * @memberOf Names
-     */
-    fetch(){
-        apiCall(null, 'get', 'patient/' + this.props.uuid + '?v=full').then((res) => {
-          this.setState({   res: res.person,
-              givenName: res.person.preferredName.givenName,
-              middleName: res.person.preferredName.middleName,
-              familyName: res.person.preferredName.familyName,
-              voided: res.person.preferredName.voided,
-              creator: res.person.auditInfo.creator.display,
-              dateCreated: res.person.auditInfo.dateCreated
-            });
-        });
+  constructor(props) {
+    super(props);
+    this.state = {
+      display: 'view',
+      givenName: '',
+      middleName: '',
+      familyName: '',
+      voided: false,
+      preferred: true,
+      creator: '',
+      dateCreated: '',
+      response: '',
+      errorGivenName: '',
+      errorFamilyName: ''
     };
-
-    /**
-     * handleEdit - changes display state to view to enable editing of input elements
-     *@param {object} event - event object
-     *
-     * @memberOf Name
-     */
-    handleEdit(){
-        this.setState({display: 'edit'});
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.fetch = this.fetch.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetch();
+  };
+
+  fetch() {
+    apiCall(null, 'get', 'patient/' + this.props.uuid + '?v=full').then((response) => {
+      this.setState({
+        response: response.person,
+        givenName: response.person.preferredName.givenName,
+        middleName: response.person.preferredName.middleName,
+        familyName: response.person.preferredName.familyName,
+        voided: response.person.preferredName.voided,
+        creator: response.person.auditInfo.creator.display,
+        dateCreated: response.person.auditInfo.dateCreated,
+      });
+    });
+  };
+
+  handleEdit() {
+    this.setState({ display: 'edit' });
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target
+    event.preventDefault();
+    this.setState({
+      [name]: value
+    });
+  }
+
+  handleSave() {
+    this.setState((prevState) => ({
+      givenName: this.refs.givenname.value,
```

Comment:
If I'm right I don't think you need to set state here again because `this.handleChange` should have taken care of that for you.